.github/workflows/skywalking.yaml

```diff
@@ -568,22 +607,26 @@ jobs:
     timeout-minutes: 10
     steps:
       - name: Merge Requirement
+        # check changes, sanity, dependency license, unit, integration, e2e, e2e-istio and e2e-java-versions, 
+        # if all of them are working as expected then naturally exits else return error code
         run: |
-          if [[ ${{ needs.dependency-license.result }} != 'success' ]]; then
-            exit -1
-          fi
-          if [[ ${{ needs.unit-test.result }} != 'success' ]]; then
-            exit -1
-          fi
-          if [[ ${{ needs.integration-test.result }} != 'success' ]]; then
-            exit -1
-          fi
-          if [[ ${{ needs.e2e-test.result }} != 'success' ]]; then
-            exit -1
-          fi
-          if [[ ${{ needs.e2e-test-istio.result }} != 'success' ]]; then
-            exit -1
-          fi
-          if [[ ${{ needs.e2e-test-java-versions.result }} != 'success' ]]; then
-            exit -1
-          fi
+          execute=${{ needs.changes.outputs.oap }}
```

Comment:
I’d just write if not executed exit 0, and check the remaining one by one so you don’t need to check execute there too

Reply:
I remember when I first experimented weeks ago, it has to be like this because when there's something wrong before execute, the `changes` job will not run normally.

That said this variable won't be populated but there could be failed jobs before(including changes itself). Then I thought we could solve this by not `need` the sanity check in `changes`, but that in fact also doesn't work because changes -> skip[s] -> required check runs faster than the parallel sanity check, the button will be green before sanity check finishes. so we need to add another condition to check sanity check results to the `if always()`, 

This probably should work, but `changes` itself could potentially fail or suffer from other problems, so I chose to check each `skip` status to ensure the pipeline is doing correctly. Otherwise, it's may become flaky.